src/main/java/it/gov/pagopa/payhub/auth/exception/AuthExceptionHandler.java

```diff
@@ -19,20 +18,36 @@
 @Order(Ordered.HIGHEST_PRECEDENCE)
 public class AuthExceptionHandler {
 
-    @ExceptionHandler(InvalidTokenException.class)
+    @ExceptionHandler({InvalidTokenException.class, TokenExpiredException.class})
     @ResponseStatus(HttpStatus.UNAUTHORIZED)
-    public AuthErrorDTO handleInvalidTokenException(InvalidTokenException ex, HttpServletRequest request){
+    public AuthErrorDTO handleInvalidGrantError(RuntimeException ex, HttpServletRequest request){
         String message = logAndReturnUnauthorizedExceptionMessage(ex, request);
 
         return new AuthErrorDTO(AuthErrorDTO.ErrorEnum.INVALID_GRANT, message);
     }
 
-    @ExceptionHandler(TokenExpiredException.class)
+    @ExceptionHandler(InvalidExchangeClientException.class)
     @ResponseStatus(HttpStatus.UNAUTHORIZED)
-    public AuthErrorDTO handleTokenExpiredException(TokenExpiredException ex, HttpServletRequest request){
+    public AuthErrorDTO handleInvalidClientError(RuntimeException ex, HttpServletRequest request){
         String message = logAndReturnUnauthorizedExceptionMessage(ex, request);
 
-        return new AuthErrorDTO(AuthErrorDTO.ErrorEnum.INVALID_GRANT, message);
+        return new AuthErrorDTO(AuthErrorDTO.ErrorEnum.INVALID_CLIENT, message);
+    }
+
+    @ExceptionHandler({InvalidExchangeRequestException.class, InvalidTokenIssuerException.class})
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public AuthErrorDTO handleInvalidRequestError(RuntimeException ex, HttpServletRequest request){
+        String message = logAndReturnUnauthorizedExceptionMessage(ex, request);
```

Comment:
the log method print 401 on std out even tho the response is 400. Is it intentional?